python/ray/serve/tests/test_gradio_visualization.py

```diff
@@ -234,29 +234,40 @@ async def test_gradio_visualization_e2e(graph1):
     handle = serve.run(DAGDriver.bind(dag))
     visualizer = GraphVisualizer()
     (_, url, _) = visualizer.visualize_with_gradio(handle, _launch=True, _block=False)
-
-    async with aiohttp.ClientSession() as session:
-
-        async def fetch(data, fn_index):
-            async with session.post(
-                f"{url.strip('/')}/api/predict/",
-                json={
-                    "session_hash": "random_hash",
-                    "data": data,
-                    "fn_index": fn_index,
-                },
-            ) as resp:
-                return (await resp.json())["data"]
-
-        await fetch(
-            [random.randint(0, 100), 1, 2], 0
-        )  # sends request to dag with input (1,2)
-        values = await asyncio.gather(
-            fetch([], 1),  # fetches return value for one of the nodes
-            fetch([], 2),  # fetches return value for the other node
-        )
-
-    assert [1] in values and [2] in values
+    client = Client("http://localhost:7860")
```

Comment:
Seems like the [Python client](https://www.gradio.app/docs/client) instead of the rest api is the new API for interacting with a gradio app, changed the test to use the gradio python client

Reply:
That is a much better idea thank you 🙏